### PR TITLE
SparkServer was updated to be runnable on http and https protocols at…

### PR DESCRIPTION
**Pull request by gimlet2**
… same time. Api call was added to SparkBase to support that. Ability to specify special ssl port and justSsl flag were added.

### CODE REVIEW

**Comment by georgeyanev** [comment]
When is supposed this to be merged? I need this.

File: src/main/java/spark/SparkBase.java [hunk 1]
```diff
@@ -22,17 +21,20 @@
 public abstract class SparkBase {
     private static final Logger LOG = LoggerFactory.getLogger("spark.Spark");
     public static final int SPARK_DEFAULT_PORT = 4567;
+    public static final int SPARK_DEFAULT_PORT_SSL = 4568;
     protected static final String DEFAULT_ACCEPT_TYPE = "*/*";
 
     protected static boolean initialized = false;
 
     protected static int port = SPARK_DEFAULT_PORT;
+    protected static int sslPort = SPARK_DEFAULT_PORT;
```
Comment (georgeyanev) [inline src/main/java/spark/SparkBase.java]: Should't sslPort have the value of SPARK_DEFAULT_PORT_SSL?

**Comment by elanhami** [comment]
+1

**Comment by gimlet2** [comment]
Pull request was synced with latest code base.

**Comment by georgeyanev** [comment]
Where exactly is it synchronized? In the master code base I can not find the string 'SPARK_DEFAULT_PORT_SSL' neither the 'sslPort' variable?
And what about the wrong initialization of sslPort here: https://github.com/perwendel/spark/pull/329/commits/1e2cfa11aa25e2e4eb4c38ed4676b5abf60634fa

**Comment by gimlet2** [comment]
It was better to say that it is redone. @georgeyanev Please see commit e1b6144b63ef5346a35be3cffbf6f96134b4c784

**Comment by Thileepan** [comment]
@gimlet2 Is this feature is available in 2.5.3 release? I need this feature ASAP. thanks

**Comment by perwendel** [comment]
Is this issue still relevant now when there's an instance API?

```java
Service http = ignite()
Service https = ignite()

http.port(1234)
https.port(1235)
https.secure(...
```

**Comment by tipsy** [comment]
As per mentioned, I think this should be taken care of by the instance-api, or in a layer in front of Spark.

**Pull request closed without merging**